src/volunteer-form/index.tsx

```diff
@@ -356,13 +366,20 @@ class VolunteerForm extends React.Component<VolunteerFormProps> {
       submittingTutor:
         this.props.aspect === 'tutoring' || this.state.submittingTutor,
     });
-    await this.context.signup(this.context.user);
-    this.setState({
-      submittedMentor:
-        this.props.aspect === 'mentoring' || this.state.submittedMentor,
-      submittedTutor:
-        this.props.aspect === 'tutoring' || this.state.submittedTutor,
-    });
+
+    const result = (await this.context.signup(
```

Comment:
See my earlier comment. You should just catch the errors thrown by `this.context.signup` instead of making `this.context.signup` return an `ApiCallResult`.

I think it's better for `this.context.signup` to throw an error rather than return a result (e.g. if it just returns a result, another developer in the future might think that it works even though it really errored). Though I'm open to arguments on this. Is it better to make it return a result rather than throw an error?

Reply:
I think you're right, the solution with throwing errors looks more cleaner. I returned it. Plus, I suggest throwing a custom error type (to be more specific and to communicate HTTP response code to HttpErrorDialog).